proof-systems/src/darlin/accumulators/dlog.rs

```diff
@@ -340,9 +346,15 @@ impl<G: AffineCurve, D: Digest> ItemAccumulator for DLogItemAccumulator<G, D> {
         let accumulate_time = start_timer!(|| "Accumulate");
 
         // Initialize Fiat-Shamir rng
-        let mut fs_rng = <InnerProductArgPC<G, D> as PolynomialCommitment<G::ScalarField>>::RandomOracle::from_seed(
-            &to_bytes![&Self::PROTOCOL_NAME, ck.hash.clone(), accumulators.as_slice()].unwrap()
-        );
+        let fs_rng_init_seed = {
+            let mut seed_builder = <<InnerProductArgPC<G, D> as PolynomialCommitment<G::ScalarField>>::RandomOracle as FiatShamirRng>::Seed::new();
+            seed_builder.add_bytes(&Self::PROTOCOL_NAME)?;
+            seed_builder.add_bytes(&ck.hash)?;
+            // TODO: Shall we decompose this further when passing it to the seed builder ?
```

Comment:
What would be the reason for a decomposition?

Reply:
If we consider the accumulators as a single entity contributing to the seed, then the internal num_elements counter will just be incremented by one and we will add just one elem length, that is the total length of the accumulators; instead we can consider each accumulator singularly, so the internal num_elements counter will be incremented by accumulators.len() and accumulators.len() element will be pushed to the internal element_length array.
We will end up with two different seeds

Reply:
We assume the number of accumulators to be clear from the context. As we use constant length encoding of field elements, we may use add_bytes without producing collisions in the serialization procedure.